### PR TITLE
Feature/apcv 756

### PR DESCRIPTION
**Pull request by palatsangeetha**

### CODE REVIEW

File: ovs-persistence/src/main/java/org/dcsa/ovs/persistence/entity/LocationEntity.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+import javax.persistence.*;
+
+@Data
+@Builder
+@NoArgsConstructor
+@AllArgsConstructor
+@Setter(AccessLevel.PUBLIC)
+@Entity
+@Table(name = "location")
+public class LocationEntity {
```
Comment (douradofrazer) [inline ovs-persistence/src/main/java/org/dcsa/ovs/persistence/entity/LocationEntity.java:17]: It should be `Location` rather than `LocationEntity`, I do realize this would cause a conflict with the existing entity file, here's how you could ignore it using `excludeFilters`

```
@Configuration 
@ComponentScan(basePackages = {"com.example"}, excludeFilters={
@ComponentScan.Filter(type=FilterType.ASSIGNABLE_TYPE, value=Foo.class)})
public class MySpringConfiguration {}
```

File: ovs-persistence/src/main/java/org/dcsa/ovs/persistence/entity/TransportCall.java [hunk 2]
```diff
@@ -14,7 +14,7 @@
 @Builder
 @NoArgsConstructor
 @AllArgsConstructor
-@Setter(AccessLevel.PRIVATE)
+@Setter(AccessLevel.PUBLIC)
```
Comment (douradofrazer) [inline ovs-persistence/src/main/java/org/dcsa/ovs/persistence/entity/TransportCall.java:17]: Is this change needed?

File: ovs-persistence/src/main/java/org/dcsa/ovs/persistence/entity/Service.java [hunk 3]
```diff
@@ -27,7 +27,7 @@
 @Builder
 @NoArgsConstructor
 @AllArgsConstructor
-@Setter(AccessLevel.PRIVATE)
+@Setter(AccessLevel.PUBLIC)
```
Comment (douradofrazer) [inline ovs-persistence/src/main/java/org/dcsa/ovs/persistence/entity/Service.java:30]: Is this change needed?

File: ovs-service/src/main/java/org/dcsa/ovs/mapping/TransportCallMapper.java [hunk 4]
```diff
@@ -79,4 +81,30 @@ PortTerminalLocation mapPortTerminalLocation(TransportCall transportCall) {
...
+        target = "importVoyage.universalVoyageReference",
+        source = "universalImportVoyageReference"),
+      @Mapping(
+        target = "exportVoyage.universalVoyageReference",
+        source = "universalExportVoyageReference"),
+      @Mapping(target = "portCallStatusCode", source = "statusCode")
+    })
+  public abstract TransportCall toEntity(TransportCallTO transportCallTO);
+
+  LocationEntity mapLocation(PortTerminalLocation location) {
```
Comment (douradofrazer) [inline ovs-service/src/main/java/org/dcsa/ovs/mapping/TransportCallMapper.java:100]: There are 4 location types, correct? Is some mapping missing?

**Review by douradofrazer — CHANGES_REQUESTED** [review]

File: ovs-persistence/src/main/java/org/dcsa/ovs/persistence/entity/Vessel.java [hunk 5]
```diff
@@ -11,7 +11,7 @@
 @Builder
 @NoArgsConstructor
 @AllArgsConstructor
-@Setter(AccessLevel.PRIVATE)
+@Setter(AccessLevel.PUBLIC)
```
[the remark this answers is not in the document]
Comment (palatsangeetha) [reply ovs-persistence/src/main/java/org/dcsa/ovs/persistence/entity/Vessel.java:14]: Had to set vesselOperatorCarrier explicitly from the service class. For that made it public.

File: ovs-persistence/src/main/java/org/dcsa/ovs/persistence/entity/TransportCall.java (hunk 2 above)
Comment (palatsangeetha) [reply ovs-persistence/src/main/java/org/dcsa/ovs/persistence/entity/TransportCall.java:17]: Had to set vessel explicitly to Transport call.

File: ovs-service/src/main/java/org/dcsa/ovs/mapping/TransportCallMapper.java (hunk 4 above)
Comment (palatsangeetha) [reply ovs-service/src/main/java/org/dcsa/ovs/mapping/TransportCallMapper.java:100]: Will add it.

**Review by douradofrazer — CHANGES_REQUESTED** [review]

File: ovs-service/src/main/java/org/dcsa/ovs/service/unofficial/UnofficialServiceScheduleService.java [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+import java.util.stream.Stream;
+
+@Service
+@RequiredArgsConstructor
+public class UnofficialServiceScheduleService {
+
+  private final ServiceRepository serviceRepository;
+  private final ServiceScheduleMapper serviceScheduleMapper;
+  private final CarrierRepository carrierRepository;
+  private static List STATUS_CODE =
```
[the remark this answers is not in the document]
Comment (palatsangeetha) [reply ovs-service/src/main/java/org/dcsa/ovs/service/unofficial/UnofficialServiceScheduleService.java:27]: Should I validate the input string value against the values in the enum?

File: ovs-transfer-obj/src/main/java/org/dcsa/ovs/transferobjects/ServiceScheduleTO.java [hunk 7]
```diff
@@ -8,6 +9,8 @@ public record ServiceScheduleTO (
   String carrierServiceName,
   String carrierServiceCode,
   String universalServiceReference,
+  @JsonProperty(access = JsonProperty.Access.WRITE_ONLY)
+  String serviceOwnerCode,
```
[the remark this answers is not in the document]
Comment (palatsangeetha) [reply ovs-transfer-obj/src/main/java/org/dcsa/ovs/transferobjects/ServiceScheduleTO.java:13]: Will add it

**Review by douradofrazer — APPROVED** [review]
